### PR TITLE
Added Validation rule to not allow dob >= today & dob < 01-01-1940

### PR DESCRIPTION
**Pull request by MrKevJoy**
### Context

Added new validation rule to stop dob being in the future and less than 1940-01-01

### Changes proposed in this pull request

new validation rule & tests.

### Guidance to review

Inlude any useful information needed to review this change.
Inlude any dependencies that are required for this change.

### Checklist

-   [] Attach to Trello card
-   [x] Rebased master
-   [x] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

File: src/DqtApi/V2/Validators/GetOrCreateTrnRequestValidator.cs [hunk 1]
```diff
@@ -28,7 +29,14 @@ public GetOrCreateTrnRequestValidator()
                 .MaximumLength(AttributeConstraints.Contact.LastNameMaxLength);
 
             RuleFor(r => r.BirthDate)
-                .NotEmpty();
+                .NotEmpty()
+                .Custom((value, ctx) =>
+                {
+                    if (value >= DateOnly.FromDateTime(clock.UtcNow) || value <= new DateOnly(1940, 1, 1))
```
Comment (gunndabad) [inline src/DqtApi/V2/Validators/GetOrCreateTrnRequestValidator.cs:35]:
```suggestion
                    if (value >= DateOnly.FromDateTime(clock.UtcNow) || value < new DateOnly(1940, 1, 1))
```

Comment (gunndabad) [reply src/DqtApi/V2/Validators/GetOrCreateTrnRequestValidator.cs:35]: Small thing but that matches current behaviour exactly

**Comment by fofr** [comment]
What's the reasoning for 1 January 1940?

**Comment by gunndabad** [comment]
> What's the reasoning for 1 January 1940?

Not sure, we're just mirroring the existing rules for now.

**Review by gunndabad — APPROVED** [review]

**Pull request merged**